### PR TITLE
fix: Improve invalid fields

### PR DESCRIPTION
**Pull request by anhanh11001**
Details:
- Check all invalid fields and show errors on all fields

Fixes: #2239

Screenshots for the change:

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewHolder.kt [hunk 1]
```diff
@@ -217,12 +217,15 @@ class AttendeeViewHolder(private val binding: ItemAttendeeBinding) : RecyclerVie
...
+        var valid = true
         requiredList.forEach {
-            if (!it.checkEmpty() ||
-                (it.inputType == InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS && !it.checkValidEmail()) ||
-                (it.inputType == InputType.TYPE_TEXT_VARIATION_URI && !it.checkValidURI())) return false
+            if (!it.first.checkEmpty(it.second) ||
+                (it.first.inputType == InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS &&
+                    !it.first.checkValidEmail(it.second)) ||
+                (it.first.inputType == InputType.TYPE_TEXT_VARIATION_URI &&
+                    !it.first.checkValidURI(it.second))) valid = false
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewHolder.kt]: this if condition looks too complicated, why don't you simplify it with helper methods

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewHolder.kt]: updated

File: app/src/main/java/org/fossasia/openevent/general/speakercall/EditSpeakerFragment.kt [hunk 2]
```diff
@@ -384,8 +384,12 @@ class EditSpeakerFragment : Fragment(), ComplexBackPressFragment {
         return "data:image/jpeg;base64," + Base64.encodeToString(bytes, Base64.DEFAULT)
     }
 
-    private fun checkSpeakerSuccess(): Boolean =
-        rootView.speakerEmail.checkEmpty() && rootView.speakerName.checkEmpty()
+    private fun checkSpeakerSuccess(): Boolean {
+        var valid = true
+        valid = rootView.speakerEmail.checkEmpty(rootView.speakerEmailLayout) && valid
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/speakercall/EditSpeakerFragment.kt]: isn't the part after && always true ?

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/speakercall/EditSpeakerFragment.kt]: updated

**Pull request merged**